pyproject.toml

```diff
@@ -50,6 +50,10 @@ pyparsing = "^3.0.9"
 requests = "^2.28.1"
 pydantic = "^2.5.1"
 pydantic-settings = "^2.1.0"
+flask = "^3.0.3"
+requests-oauthlib = "^2.0.0"
+waitress = "^3.0.0"
+pynacl = "^1.5.0"
```

Comment:
I'm not a fan of pulling in flask and waitress. Receiving a request and responding with a redirect should be doable in Python's standard library.

Reply:
Maybe something like this?

```python
from http.server import BaseHTTPRequestHandler, HTTPServer

PORT = 5000
REDIRECT_URL = "https://www.hzdr.de"


class Handler(BaseHTTPRequestHandler):
    def do_GET(self):
        is_callback = self.path == "/callback"
        if is_callback:
            self.send_response(200)
            self.send_header("Content-Type", "text/plain")
        else:
            self.send_response(302)
            self.send_header("Location", REDIRECT_URL)
        self.end_headers()
        if is_callback:
            self.wfile.write(b"You can close this window now.")

    def log_request(self, code="-", size="-"):
        """Don't log anything."""
        pass


with HTTPServer(("127.0.0.1", PORT), Handler) as server:
    print(f"http://127.0.0.1:{PORT}", f"http://127.0.0.1:{PORT}/callback", sep="\n")
    server.serve_forever()
```

Reply:
OK, the callback needs to be in `do_POST`, but the principle is the same.

Reply:
OK, this should work for real this time 😄 

```python
from http.server import BaseHTTPRequestHandler, HTTPServer
from urllib.parse import parse_qs, urlparse

PORT = 5000
REDIRECT_URL = "https://www.hzdr.de"


class Handler(BaseHTTPRequestHandler):
    def do_GET(self):
        parsed = urlparse(self.path)
        is_callback = parsed.path == "/callback"

        if is_callback:
            self.send_response(200)
            self.send_header("Content-Type", "text/plain")
            print(parse_qs(parsed.query))
        else:
            self.send_response(302)
            self.send_header("Location", REDIRECT_URL)
        self.end_headers()
        if is_callback:
            self.wfile.write(b"You can close this window now.")

    def log_request(self, code="-", size="-"):
        pass


with HTTPServer(("127.0.0.1", PORT), Handler) as server:
    print(f"http://127.0.0.1:{PORT}", f"http://127.0.0.1:{PORT}/callback", sep="\n")
    server.serve_forever()
```

Reply:
I agree to avoid adding lots of additional deps if we can avoid it.